src/transfers-history/services/SpokePoolEventsQueryService.ts

```diff
@@ -78,7 +78,16 @@ export class SpokePoolEventsQueryService {
       return { ...acc, [event.blockNumber]: true };
     }, {} as Record<number, any>);
     const uniqueBlockNumbersList = Object.keys(uniqueBlockNumbers).map((blockNumber) => parseInt(blockNumber));
-    const blocks = await Promise.all(uniqueBlockNumbersList.map((blockNumber) => this.provider.getBlock(blockNumber)));
+    this.logger.debug(
+      "[getBlocksTimestamp]",
+      `🟢 chain ${this.chainId}: fetching ${uniqueBlockNumbersList.length} blocks`
+    );
+    const blocksChunks = this.getArrayChunks(uniqueBlockNumbersList);
+    const blocks = [];
+    for (const blocksChunk of blocksChunks) {
+      const newBlocks = await Promise.all(blocksChunk.map((blockNumber) => this.provider.getBlock(blockNumber)));
+      blocks.push(...newBlocks);
+    }
```

Comment:
fetch blocks info in batches

Reply:
I want to build a small API service which serves block metadata. Something like `https://BLOCKS_API_SERVICE/blocks?number=BLOCK_NUMBER&chainId=CHAIN_ID`. It'll store the response in a cache and all further requests will use the cache instead of hitting the web3 providers again

Reply:
that would be nice, how do you plan on doing this

Reply:
A simple ExpressJS app (eventually managed from Zion) would do it. It'll just receive the providers urls by ENV vars and store in memory the requested blocks

Reply:
Relatedly, we just created a Redis db for timestamp -> blocknumber (and a few other expensive bot queries) lookups on mainnet. It works pretty well from cloud run. We could use a similar strategy here.

@daywiss this might be something that dev support could devise a strategy for that would work for many applications.